GitCommands/Git/GitModule.cs

```diff
@@ -1098,7 +1101,7 @@ public Process RunBash(string bashCommand = null)
                 };
 
                 string args = "";
-                string cmd = termEmuCmds.FirstOrDefault(termEmuCmd => !string.IsNullOrEmpty(RunCmd("which", termEmuCmd)));
+                string cmd = termEmuCmds.FirstOrDefault(termEmuCmd => !string.IsNullOrEmpty(ThreadHelper.JoinableTaskFactory.Run(() => RunCmdAsync("which", termEmuCmd))));
```

Comment:
What do you think about the approach @drewnoakes applied in #4501:

>Approach:
>- Find a stack that's commonly blocking a thread pool (worker) thread
>- Find a the lowest-level sync method that should be made async
>- Duplicate it, making the copy async
>- Mark the original `[Obsolete]`
>- Walk up the stack, making callers async using the same process described above

The original implementation can call the async one wrapped in JTF.Run.
This way the PR's would produce smaller diffs with less noise, in the end `ThreadHelper.JoinableTaskFactory.Run(() => RunCmdAsync("which", termEmuCmd)))` will be turned in an async call. Does it make sense to touch it twice?

Reply:
In the end, the breadth-first approach will take us to the same place as a depth-first approach. I like this one because it affects fewer methods at a time, and doesn't require you to verify that copied methods have the correct content and are kept in sync as other changes are made.